tls/s2n_cipher_preferences.c

```diff
@@ -1850,6 +1850,45 @@ const struct s2n_cipher_preferences cipher_preferences_pq_tls_1_0_2021_05_26 = {
     .allow_chacha20_boosting = false,
 };
 
+/* Same as 2021_05_26, but:
+ * 1. TLSv1.2 Kyber KEM cipher suites are removed
+ * 2. AES 256 is preferred over AES 128 as in the cloudfront ciphersuites
```

Comment:
This line is only partially correct. AES 256 is preferred over AES 128 **when TLS 1.3 is negotiated**, but if TLS 1.2 is used then AES 128 is preferred for 256. Wanted to double check if this was intended?

Reply:
I believe so. Inferring from the other policies, it looks like we generally prefer AES 128 for non-PQ (presumably for performance reasons) and **always** prefer AES 256  for hybrid PQ for security reasons. So, this cipher preference guarantees that if PQ is negotiated (implying TLS 1.3), AES 256 will be used. Perhaps in an ideal world, we would negotiate AES 128 in the non-PQ TLS 1.3 case for performance reasons, but I don't believe that the current security policy + cipher preferences abstraction supports that level of differentiation (i.e. making the choice of cipher preference based on which TLS version actually ends up getting negotiated).

Below is some quick CLI testing. Here's the server invocation:

```
$ ./build/bin/s2nd -X --self-service-blinding --non-blocking --key ./tests/pems/rsa_4096_sha512_client_key.pem --cert ./tests/pems/rsa_4096_sha512_client_cert.pem -t ./tests/pems/rsa_4096_sha512_client_cert.pem -c PQ-TLS-1-3-2023-06-01 -T localhost 19013
...
```

PQ TLS 1.3:

```
$ ./build/bin/s2nc --non-blocking -e -T --insecure -c PQ-TLS-1-3-2023-06-01 localhost 19013 | egrep 'Actual|KEM|Cipher'
Actual protocol version: 34
KEM: NONE
KEM Group: secp768r1_kyber-768-r3
Cipher negotiated: TLS_AES_256_GCM_SHA384
```

non-PQ TLS 1.3:

```
$ ./build/bin/s2nc --non-blocking -e -T --insecure -c default_tls13 localhost 19013 | egrep 'Actual|KEM|Cipher'
Actual protocol version: 34
KEM: NONE
KEM Group: NONE
Cipher negotiated: TLS_AES_256_GCM_SHA384
```

PQ TLS 1.2 (all TLS 1.2 negotiated against the new policy `PQ-TLS-1-3-2023-06-01` is non-PQ):

```
$ ./build/bin/s2nc --non-blocking -e -T --insecure -c default localhost 19013 | egrep 'Actual|KEM|Cipher'
Actual protocol version: 33
KEM: NONE
KEM Group: NONE
Cipher negotiated: ECDHE-RSA-AES128-GCM-SHA256
```

Thank you for clarification. I'll update the comment.